server/service/src/sync/sync_status/logger.rs

```diff
@@ -209,9 +220,210 @@ impl<'a> SyncLogger<'a> {
             }
         };
 
-        self.sync_log_repo
-            .upsert_one(&self.row)
-            .map_err(SyncLoggerError)?;
+        self.sync_log_repo.upsert_one(&self.row)?;
         Ok(())
     }
 }
+
+impl SyncLogError {
+    /// Map SyncError to SyncLogError, to be queried later and translated in front end
+    fn from_sync_error(sync_error: &SyncError) -> Self {
+        let sync_api_error = match &sync_error {
+            // Sync Api Error
+            SyncError::CentralPullError(CentralPullError::SyncApiError(error))
+            | SyncError::RemotePullError(RemotePullError::SyncApiError(error))
+            | SyncError::PostInitialisationError(PostInitialisationError(error))
+            | SyncError::RemotePushError(RemotePushError::SyncApiError(error))
+            | SyncError::WaitForIntegrationError(WaitForIntegrationError::SyncApiError(error)) => {
+                error
+            }
+            // Integration timeout reached
+            SyncError::WaitForIntegrationError(_) => {
+                return Self::new(SyncLogRowErrorCode::IntegrationTimeoutReached, sync_error)
+            }
+            // Internal errors
+            _ => return Self::message_only(sync_error),
```

Comment:
Still a chance to miss an error here, similar to error downcasting, but this syntax and logic I think is clearer